Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py

```diff
@@ -243,11 +256,19 @@ def fetch_incidents(client, last_run, first_fetch_time, event_type_filter, threa
 
             if raw_event["threatTime"] > fetch_time:
                 last_fetch = raw_event["threatTime"]
-
+            incident['Occurred'] = raw_event["clickTime"] if raw_event["clickTime"] > raw_event["threatTime"] else \
+                raw_event["threatTime"]
             incidents.append(incident)
-    last_fetch_datetime = get_now()
-    next_run = {'last_fetch': last_fetch_datetime}
 
+    last_fetch = last_fetch[:-5] + 'Z' if last_fetch[-5] == '.' else last_fetch
+    last_fetch_datetime = datetime.strptime(last_fetch, DATE_FORMAT)
+    last_fetch = (last_fetch_datetime + timedelta(seconds=1)).strftime(DATE_FORMAT)
+    next_run = {'last_fetch': last_fetch}
+
+    # limit incidents to the limit given
+    incidents.sort(key=lambda a: a.get('Occurred'))
+    if len(incidents) > limit:
+        incidents = incidents[:limit]
```

Comment:
the last_fetch must be the last_fetch of the 50th incident

Reply:
so you should delete all the places where you set the last_Fetch and take the last_fetch from the `occurred` field of an last (50th ) incident